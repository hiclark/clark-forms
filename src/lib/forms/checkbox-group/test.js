import React from 'react';

import CheckboxGroup from './';

describe('<CheckboxGroup />', () => {
  const props = {
    onCheckboxClick: jest.fn(),
    checked: true,
    name: 'Name',
    index: 1,
    value: false,
  };

  describe('when checked is true', () => {
    const wrapper = shallow(<CheckboxGroup {...props} />);

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with label', () => {
    const propsWithLabel = { ...props, label: 'Label' };
    const wrapper = shallow(<CheckboxGroup {...propsWithLabel} />);

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when checked is false', () => {
    const propsWithLabel = { ...props, checked: false };
    const wrapper = shallow(<CheckboxGroup {...propsWithLabel} />);

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
