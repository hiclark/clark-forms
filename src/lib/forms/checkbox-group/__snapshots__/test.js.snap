// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxGroup /> when checked is false matches the snapshot 1`] = `
<Field
  checked={false}
  component={[Function]}
  index={1}
  name="Name"
  onCheckboxClick={[Function]}
  value={false}
/>
`;

exports[`<CheckboxGroup /> when checked is true matches the snapshot 1`] = `
<Field
  checked={true}
  component={[Function]}
  index={1}
  name="Name"
  onCheckboxClick={[Function]}
  value={false}
/>
`;

exports[`<CheckboxGroup /> with label matches the snapshot 1`] = `
<Field
  checked={true}
  component={[Function]}
  index={1}
  label="Label"
  name="Name"
  onCheckboxClick={[Function]}
  value={false}
/>
`;
